packages/components/src/public/QueryModel/withQueryModels.tsx

```diff
@@ -56,6 +56,55 @@ export function withSearchParams<T>(Component: WithSearchParamsComponent<T>): Co
     return Wrapped;
 }
 
+function columnHasFilter(fieldKey, filters: Filter.IFilter[]): boolean {
+    return filters.some(filter => filter.getColumnName() === fieldKey);
+}
+
+function columnsHaveFilter(columns: string[], filters: Filter.IFilter[]): boolean {
+    return columns.some(fieldKey => columnHasFilter(fieldKey, filters));
+}
+
+export enum ChangeType {
+    add = 'add',
+    delete = 'delete',
+    update = 'update',
+}
+
+interface BaseModelChange {
+    changeType: ChangeType;
+}
+
+export interface AddChange extends BaseModelChange {
+    changeType: ChangeType.add;
+}
+
+/**
+ * selectionsForReplace: an optional set of row keys to select after the model is reset.
+ */
+export interface DeleteOptions {
+    selectionsForReplace?: string[];
```

Comment:
I remember us chatting about these selections being an array of string vs number. Any chance this can/should be number[]?

Reply:
Just wanted to follow up here in a way that would record our conversation, but as we found out yesterday it's critically important that selection keys are always typed as string, because they represent the keys in the QueryModel.rows object. If you pass numbers they will not work, and instead will only lead to silent failures. I found a bug that was caused by this exact problem, which I fixed in [this commit](https://github.com/LabKey/labkey-ui-components/pull/1726/commits/52497c7c18c60c4e882edaabae905974a9a6cf5f).